test/sql/window/test_window_distinct.test

```diff
@@ -0,0 +1,94 @@
+# name: test/sql/window/test_window_distinct.test
+# description: Windowed distinct aggregates
+# group: [window]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT COUNT(DISTINCT 42) OVER ()
+----
+1
+
+statement ok
+WITH t AS (SELECT col0 AS a, col1 AS b FROM (VALUES(1,2),(1,1),(1,2),(2,1),(2,1),(2,2),(2,3),(2,4)) v) SELECT *, COUNT(b) OVER(PARTITION BY a), COUNT(DISTINCT b) OVER(PARTITION BY a) FROM t;
+
+# Tests from the paper
+statement ok
+CREATE TABLE figure1 AS 
```

Comment:
Could we add some more distinct tests? e.g.:

* Distinct over nested types, e.g. LIST/STRUCT
* DISTINCT aggregate with NULL values in the dataset
* DISTINCT aggregate over a larger data set (e.g. lineitem or so)
* DISTINCT with RANGE instead of ROWS perhaps?
* DISTINCT with an aggregate with a destructor (e.g. `LIST` or `STRING_AGG`)
* Maybe `MEDIAN` distinct, or distinct for aggregates that have a special `window` function?